.storybook/recipes/Filters/Filters.tsx

```diff
@@ -37,10 +36,167 @@ export const Filters = ({ placement, ...other }: Props) => {
       setIsLarge(false);
     }
   }, 200);
+
+  const [checked, setChecked] = useState({
+    salad: false,
+    sandwich: false,
+    soup: false,
+  });
+
+  const onCheckboxChange = (food: string) => {
+    const newChecked = { ...checked };
+    newChecked[food] = !checked[food];
+    setChecked(newChecked);
+  };
+
+  /**
+   * Counts how many filters have been applied.
+   */
+  const filterCount = Object.values(checked).reduce((count: number, status) => {
+    if (status) return count + 1;
+    else return count;
+  }, 0);
+
+  const hasSelectedFilters = filterCount > 0;
+  const triggerText = hasSelectedFilters
+    ? `Filters (${filterCount})`
+    : 'Filters';
+
+  const checkboxes = (
+    <FiltersCheckboxField legend="Filters Segment 1">
+      <Checkbox
+        checked={checked.soup}
+        label="Soup"
+        onChange={() => onCheckboxChange('soup')}
+      />
+      <Checkbox
+        checked={checked.salad}
+        label="Salad"
+        onChange={() => onCheckboxChange('salad')}
+      />
+      <Checkbox
+        checked={checked.sandwich}
+        label="Sandwich"
+        onChange={() => onCheckboxChange('sandwich')}
+      />
+    </FiltersCheckboxField>
+  );
+
+  const foods = [
+    {
+      name: 'Cereal',
+      isSoup: true,
+      isSalad: true,
+      isSandwich: false,
+    },
+    {
+      name: 'Fried Rice',
+      isSoup: false,
+      isSalad: true,
+      isSandwich: false,
+    },
+    {
+      name: 'Hot Dog',
+      isSoup: false,
+      isSalad: false,
+      isSandwich: true,
+    },
+    {
+      name: 'Mashed Potatoes',
+      isSoup: true,
+      isSalad: true,
+      isSandwich: false,
+    },
+    {
+      name: 'Nigiri',
+      isSoup: false,
+      isSalad: true,
+      isSandwich: true,
+    },
+    {
+      name: 'Oatmeal',
+      isSoup: true,
+      isSalad: false,
+      isSandwich: false,
+    },
+    {
+      name: 'Soup in Bread Bowl',
+      isSoup: true,
+      isSalad: false,
+      isSandwich: true,
+    },
+    {
+      name: 'Sloppy Joe',
+      isSoup: true,
+      isSalad: true,
+      isSandwich: true,
+    },
+  ];
+
+  const filteredFoods = foods
+    .filter((food) => (checked.salad ? checked.salad === food.isSalad : true))
+    .filter((food) => (checked.soup ? checked.soup === food.isSoup : true))
+    .filter((food) =>
+      checked.sandwich ? checked.sandwich === food.isSandwich : true,
+    );
```

Comment:
For some reason I found this block really hard to understand, I guess because you have a filter for each option even when that option isn't active. I feel like it might be easier to reason about if we looped through the selected filters and filtered based on those, but the code would probably be more complicated and therefore harder to read. I dunno. Just putting it out there.

Reply:
hmmm would it be more readable if I wrote it as
```ts
const filteredFoods = (!checked.salad && !checked.soup && !checked.sandwich) || 
  .filter((food) => (checked.salad === food.isSalad))
  .filter((food) => (checked.soup === food.isSoup))
  .filter((food) => (checked.sandwich === food.isSandwich)),
);
```

Reply:
Yah this is tricky to grok at a glance. makes me think a comment would be useful regardless...

If someone were to apply this recipe but had 10 booleans to check, including a lengthy boolean at the beginning would be pretty hard to read as well. 

a little utility function might clarify things. Imagine:

```
// better names! (e.g., whenEquivalent?)
function when(filter, data) {
    return function() { filter ? filter === data : true };
}
```

Then the code would read like:

```
const filteredFoods = foods
    .filter(when(checked.salad, food.isSalad))
    .filter(when(checked.soup, food.isSoup))
    .filter(when(checked.sandwich, food.isSandwich))
```

I might have messed up the syntax but hopefully the idea makes sense!

Reply:
I like the use of a utility function and I'll also sprinkle some comments